pubtools/pulplib/_impl/client/client.py

```diff
@@ -565,3 +565,18 @@ def map_404_to_none(exception):
         )
 
         return f_map(response, error_fn=map_404_to_none)
+
+    def _do_sync(self, repo_id, sync_options):
+        if not sync_options["feed"]:
```

Comment:
Should we have this check actually?

1. Like any other options, it's possible for feed to be stored on the importer config itself, in which case it's not necessary to pass it.  Ideally the library doesn't block this scenario.
2. If the Pulp server really needs a feed (because it's missing from importer config), it should already be returning a meaningful error message, so we shouldn't have to duplicate the same error check here.

Reply:
Well, I wanted to do this "user-friendly" and fails asap if there's obvious error. As you said feed can be used from importer but to do that I would have to implement the model and another support code, which is something I'm not that much willing to do now due lots of changes needed in another projects right now. So I thought we still override feed everytime so let's check that feed is not empty here and in the future we can implement importer model in repository model.
If workflow is: get repository model, run sync, then at that time you will have already information about importer configuration and you can decide if there's feed or not and therefore if "external" feed is required. 
Another reason is log message few lines bellow this one which tells you: Syncing repository with feed: xxx.
But maybe it's unnecessary complicated and I should rather let pulp return error here.

Reply:
I didn't mean that this library needs to fetch the importer and use it to initialize feed - I meant that you can just do a sync request to Pulp without including any feed, and it'll work if the importer already has a feed or it'll return a meaningful error if it doesn't.  So you wouldn't have to add Importer models yet.

In any case though, I don't consider this a blocker for the merge request, seeing as it can be easily changed later without breaking compatibility.